streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -1470,13 +1470,15 @@ object ZChannel {
     f: (OutDone, OutDone) => OutDone
   )(implicit trace: Trace): ZChannel[Env, InErr, InElem, InDone, OutErr, OutElem, OutDone] =
     unwrapScoped[Env] {
-      ZIO.withChildren { getChildren =>
+      {
         for {
           input         <- SingleProducerAsyncInput.make[InErr, InElem, InDone]
           queueReader    = ZChannel.fromInput(input)
           n             <- ZIO.succeed(n)
           bufferSize    <- ZIO.succeed(bufferSize)
           mergeStrategy <- ZIO.succeed(mergeStrategy)
+          childFibers   <- Ref.make[Map[FiberId, Fiber[Any, Unit]]](Map.empty)
```

Comment:
Replaced `withChildren` with an explicitly tracked map of fibers, so we only interrupt the ones we spawn and not all the fibers created by the channel executor / merge implementation itself.